From 878fb46a571835137e32a163daaa68a8690ebd2f Mon Sep 17 00:00:00 2001
From: Hall Liu <hallliu@google.com>
Date: Wed, 25 Jul 2018 18:49:59 -0700
Subject: [PATCH 14/70] Add plumbing for the 911 redial

Provide a mechanism to swap out the original connection when the lower
layers trigger a redial.

Bug: 111235453
Test: manual (modify IMS stack locally to always return the 380 error
when dialing in airplane mode)

Change-Id: Ie1cab49f040dd2b34b775bd9e5d7f97fbf7aafa9
---
 src/com/android/services/telephony/TelephonyConnection.java | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/com/android/services/telephony/TelephonyConnection.java b/src/com/android/services/telephony/TelephonyConnection.java
index 2b0f49fb5..db4cc1018 100644
--- a/src/com/android/services/telephony/TelephonyConnection.java
+++ b/src/com/android/services/telephony/TelephonyConnection.java
@@ -569,6 +569,12 @@ abstract class TelephonyConnection extends Connection implements Holdable {
             updateConnectionProperties();
             sendRttSessionRemotelyTerminated();
         }
+
+        @Override
+        public void onOriginalConnectionReplaced(
+                com.android.internal.telephony.Connection newConnection) {
+            setOriginalConnection(newConnection);
+        }
     };
 
     protected com.android.internal.telephony.Connection mOriginalConnection;
-- 
2.17.1

